### PR TITLE
[MRG] use mne-bids for binder

### PR DESCRIPTION
**Pull request by sappelhoff**
potentially replaces #300 

Here I simply add a new requirements file specifically for the binder installation.

The idea would be to merge this into `maint/0.3` and see if it makes our examples work.

If it does, we can do the same for `master`.

Note: Merging this SHOULD trigger an update of the website, see: https://github.com/mne-tools/mne-bids/blob/5809a8204f0fc9f18e772807312ecfd656a6d6d9/.circleci/config.yml#L67-L72

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"
- [ ] Commit history does not contain any merge commits

### CODE REVIEW

**Comment by sappelhoff** [comment]
okay I need to do a bit more work here later and also update the release protocol

... whenever we do a release, we should change the "testing against master" etc. parts against testing against stable.

Otherwise we'll be testing mne-bids 0.3 against mne 0.XX.devX ... although it's been written to work well with mne 0.18.

**Comment by sappelhoff** [comment]
I added this part to the release protocol:

- Make sure the CI passes on master
    - If you are using the "master"/"dev" version for external packages in your CIs (e.g., `mne`, `bids-validator`), you will need to adjust this to point to a specific commit or stable version.
    - Else this will lead to problems in the future, when an old release of `mne-bids` will still use the `master` version in the CIs, for building the docs, etc.
    - Files to look at are:
        - `.travis.yml`
        - `circleci/config.yml`
        - `appveyor.yml`
        - `environment.yml`

File: .travis.yml [hunk 1]
```diff
@@ -6,9 +6,9 @@ language: python
 env:
   global:
     # can be "stable", or anything that can be used with git checkout
-    - VALIDATOR_VERSION="master"
+    - VALIDATOR_VERSION="9d8a6ffae5e0ee117b97bc8c978fe34ec3c13af4"
```
Comment (sappelhoff) [inline .travis.yml]: this points to the 1.3.0 release of bids-validator, 7 days before we released mne-bids: https://github.com/bids-standard/bids-validator/commit/9d8a6ffae5e0ee117b97bc8c978fe34ec3c13af4

File: appveyor.yml [hunk 2]
```diff
@@ -36,7 +36,7 @@ install:
   - "powershell ci-helpers/appveyor/install-miniconda.ps1"
   - "SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%"
   - "activate test"
-  - "conda install numpy scipy"
+  - "conda install numpy'<=1.14' scipy'<=0.18.1'"
```
Comment (sappelhoff) [inline appveyor.yml]: I'll try to fix appveyor in master for mne-bids 0.4 --> so that we can use the env.yml there as well.

**Comment by sappelhoff** [comment]
I get two interesting errors ...

- typeError: get_optparser() got an unexpected keyword argument 'prog_prefix'
- TypeError: loadtxt() got an unexpected keyword argument 'encoding'

unfortunately one of them kills the CircleCI flow and no docs are being build.

Any clue how to solve them?

**Comment by sappelhoff** [comment]
@jasmainak can you review and merge if you are happy, please?

I am excited to see whether this is enough to make our binder examples work :-)

File: .circleci/config.yml [hunk 3]
```diff
@@ -31,7 +31,7 @@ jobs:
                   conda update --yes --quiet conda
                   conda env create -f environment.yml --name=testenv
                   source activate testenv
-                  pip install -U https://api.github.com/repos/mne-tools/mne-python/zipball/master
+                  pip install mne==0.19
```
Comment (agramfort) [inline .circleci/config.yml]:
```suggestion
                  pip install mne>=0.19
```

you should be using 0.19.2 ie minor bug fix releases

Comment (sappelhoff) [reply .circleci/config.yml]: done, I am using `mne~=0.19.0` now, [which will always install the highest bugfix release](https://www.python.org/dev/peps/pep-0440/#compatible-release)

**Comment by jasmainak** [comment]
Okay let's give this a try! @sappelhoff let me know if it fixes binder :-)

**Comment by sappelhoff** [comment]
The website was updated: 
image

aaaand Binder .... does not work:

image

:-(

**Comment by sappelhoff** [comment]
I pushed a commit directly to see whether it's because I named the file `binder-requirements` instead of just `requirements`: https://github.com/mne-tools/mne-bids/commit/a37bb593623c07848e4912aee5cdbe4103e6afae

Let's see if that fixes it.

**Comment by sappelhoff** [comment]
Also didn't work ... next iteration: Try a distinct env.yml file: https://github.com/mne-tools/mne-bids/commit/acd5717a8c3d6c875693e53c8619d77e4f04906e

If this doesn't work either, I am lost.

**Comment by jasmainak** [comment]
@sappelhoff can we try to understand what is going on here? The binder integration with `sphinx_gallery` is quite painful in my experience. I complained about some issues here: https://github.com/sphinx-gallery/sphinx-gallery/issues/483. If there is something that can be improved in terms of documentation etc., I think it would be very welcome PR to `sphinx-gallery`.

**Pull request merged**